Aio.Multidataset.Rest.Connector/DatasetSampler.cs

```diff
@@ -1,102 +1,152 @@
+                responseContent = await response.Content.ReadAsStringAsync(cancellationToken);
+                _logger.LogInformation("Received data for data point \"{dataPointName}\" in dataset \"{datasetName}\": {data}", key, dataset.Name, responseContent);
             }
-            catch (Exception httpEx)
+            catch (Exception e)
             {
-                _logger.LogError(httpEx, "Failed to fetch data from HTTP endpoint {0}", dataSourceUrl);
-                // Return error information instead of crashing
-                httpData = $"{{\"error\": \"Failed to fetch data from {dataSourceUrl}\", \"message\": \"{httpEx.Message}\"}}";
+                _logger.LogError(e, "Failed to fetch data for data point \"{dataPointName}\" in dataset \"{datasetName}\" from source \"{dataSource}\": {errorMessage}", key, dataset.Name, dataSourceUrl, e.Message);
+                responseContent = $"{{\"error\": \"Failed to fetch data from source {dataSourceUrl}\", \"message\": \"{e.Message}\"}}";
             }
-            
-            // Create JSON response with the actual fetched data
+
             var jsonResponse = new
             {
-                currentAmbientTemperature = new
-                {
-                    dataSource = dataSourceUrl,
-                    data = httpData,
-                    timestamp = DateTimeOffset.UtcNow.ToString("o")
-                }
+                dataSource = dataSourceUrl,
+                data = responseContent,
+                timestamp = DateTimeOffset.UtcNow.ToString("o")
             };
-            
-            var jsonString = JsonSerializer.Serialize(jsonResponse);
-            _logger.LogInformation("Returning sampled data: {0}", jsonString);
-            
-            return Encoding.UTF8.GetBytes(jsonString);
+
+            try
+            {
+                sampledData.Add(key, Regex.Unescape(JsonSerializer.Serialize(jsonResponse)));
+            }
+            catch (NotSupportedException e)
+            {
+                _logger.LogError(e, "Failed to serialize response {responseContent}: {errorMessage}", responseContent, e.Message);
+            }
+        }
+
+        string jsonString = Regex.Unescape(JsonSerializer.Serialize(sampledData, _jsonSerializerOptions));
+        _logger.LogInformation("Returning sampled data: {jsonString}", jsonString);
+        return Encoding.UTF8.GetBytes(jsonString);
+    }
+
+    /// <summary>
+    /// Gets the sampling interval for the given <see cref="AssetDataset"/>.
+    /// Note that the dataset can only have one sampling interval for all of its data points.
+    /// </summary>
+    /// <param name="dataset"></param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException">Thrown, if the dataset has no data points.</exception>
+    public Task<TimeSpan> GetSamplingIntervalAsync(AssetDataset dataset, CancellationToken cancellationToken = default)
```

Comment:
I need to move sampling interval  at asset level i.e.  in datasetConfiguration instead of datapoints level as polling by Akri framework happens at dataset level only.

Reply:
?

Reply:
I wanted to ideally make this change earlier this week but away from desk (due to a work trip). The change is to use datasetconfig and not datapointconfig for getting sampling interval.